src/main/java/com/aol/cyclops/types/Value.java

```diff
@@ -49,6 +51,29 @@
 @FunctionalInterface
 public interface Value<T> extends Supplier<T>, Foldable<T>, Convertable<T>, Publisher<T>, Predicate<T>, Zippable<T> {
 
+	/**
+     * Perform a flatMap operation that will work for Value and take the first value from the Iterable returned.
+     * 
+     * @param fn FlatMaping function
+     * @return Value with flattening transformation
+     */
+    default <R> Value<R> flatMapFirst(Function<? super T, ? extends Iterable<? extends R>> fn) {
```

Comment:
Hey will, I think the requirement is wrong here. It probably should be in MonadicValue where we can make use of the existing flatMap method and extract the first value from the result. 

e.g.  something like this ->

       public <R> Eval<R> flatMapFirst(Function<? super T, ? extends MonadicValue<? extends R>> mapper) {
               return this.flatMap(a->extractFirst(mapper.apply(a)));

            }

that way the execution remains lazy. We'd also have to provide an implementation / override in each type that extends MonadicValue so that they have the correct return type when calling flatMapFirst